### ISSUE TITLE
CryptedFileKeyring._unlock() does not return a value, forces _init_file()

### ISSUE THREAD

**Issue by jaraco** [jaraco/keyring#71]
CryptedFileKeyring's keyring_key method includes the line

```
#!python

self._check_file() and self._unlock() or self._init_file()
```

However, self._unlock() does not return a value, causing self._init_file() to be called if the file exists and the correct password is entered.  The fix would be to return a value from _unlock, or simply rewrite that line as

```
#!python

if self._check_file():
    self._unlock()
else:
    self._init_file()
```

---
- Bitbucket: https://bitbucket.org/kang/python-keyring-lib/issue/71
- Originally reported by: Kyle Stark
- Originally created at: 2012-08-22T17:30:52.753

**Comment by jaraco** [comment]
Agreed. Looks like a good patch. It looks like you're preparing a pull request. If not, just let me know and I'll make the change.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Should be fixed in the pull request I submitted. Great library overall, thanks for your work.

---

Original comment by: Kyle Stark

**Comment by jaraco** [comment]
Fixed in 8881c7d88dff

---

Original comment by: Jason R. Coombs
